### ISSUE TITLE
Deploy 2020-10-07 failed on covers.openlibrary.org

### ISSUE THREAD

**Issue by cdrini** [internetarchive/openlibrary#3902]
### Summary

- **What is wrong?** https://covers.openlibrary.org 502s after deploying https://github.com/internetarchive/openlibrary/releases/tag/deploy-2020-10-07 .

(files: https://github.com/internetarchive/openlibrary/compare/deploy-2020-09-21...deploy-2020-10-07)

![image](https://user-images.githubusercontent.com/6251786/95402485-d9373280-08dd-11eb-8c5b-86cb947ea7be.png)

- **What caused it?**

- **What fixed it?** Temporary revert on ol-covers1 (see https://github.com/internetarchive/openlibrary/wiki/Deployment-Guide#rolling-back )

- **Followup actions:**
 

### Steps to close

1. [x] Assignment: Is someone assinged to this issue? (notetaker, responder)
2. [x] Labels: Is there an `Affects:` label applied?
3. [ ] Diagnosis: Add a description and scope of the issue
4. [ ] Updates: As events unfold, is noteable **provenance** documented in issue comments? (i.e. useful debug commands / steps / learnings / reference links)
5. [ ] "What caused it?" - please answer in summary
6. [ ] "What fixed it?" - please answer in summary
7. [ ] "Followup actions:" actions added to summary

**Comment by cdrini** [comment]
Dropping to P1 since the temporary revert fixed it, but this is blocking deploys, so still a P1.

**Comment by cdrini** [comment]
Here's the pertinent error from the logs @mekarpeles :

```
[2020-10-07 20:45:12 +0000] [29471] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/opt/openlibrary/venv/local/lib/python2.7/site-packages/gunicorn/arbiter.py", line 583, in spawn_worker
    worker.init_process()
  File "/opt/openlibrary/venv/local/lib/python2.7/site-packages/gunicorn/workers/base.py", line 129, in init_process
    self.load_wsgi()
  File "/opt/openlibrary/venv/local/lib/python2.7/site-packages/gunicorn/workers/base.py", line 138, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/opt/openlibrary/venv/local/lib/python2.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/opt/openlibrary/openlibrary/scripts/coverstore-server", line 50, in load
    from openlibrary.coverstore import server, code
  File "/opt/openlibrary/openlibrary/openlibrary/coverstore/server.py", line 9, in <module>
    from openlibrary.coverstore import config, code, archive
  File "/opt/openlibrary/openlibrary/openlibrary/coverstore/code.py", line 13, in <module>
    from openlibrary.coverstore.coverlib import save_image, read_image, read_file
  File "/opt/openlibrary/openlibrary/openlibrary/coverstore/coverlib.py", line 15, in <module>
    from openlibrary.coverstore.utils import random_string, rm_f
  File "/opt/openlibrary/openlibrary/openlibrary/coverstore/utils.py", line 12, in <module>
    from six.moves.urllib.parse import splitquery, unquote, unquote_plus
ImportError: No module named urllib.parse
[2020-10-07 20:45:12 +0000] [29471] [INFO] Worker exiting (pid: 29471)
Traceback (most recent call last):
  File "/opt/openlibrary/openlibrary/scripts/coverstore-server", line 79, in <module>
    start_gunicorn_server()
  File "/opt/openlibrary/openlibrary/scripts/coverstore-server", line 54, in start_gunicorn_server
    WSGIServer("%prog coverstore.yml --gunicorn [options]").run()
  File "/opt/openlibrary/venv/local/lib/python2.7/site-packages/gunicorn/app/base.py", line 223, in run
    super(Application, self).run()
  File "/opt/openlibrary/venv/local/lib/python2.7/site-packages/gunicorn/app/base.py", line 72, in run
    Arbiter(self).run()
  File "/opt/openlibrary/venv/local/lib/python2.7/site-packages/gunicorn/arbiter.py", line 232, in run
    self.halt(reason=inst.reason, exit_status=inst.exit_status)
  File "/opt/openlibrary/venv/local/lib/python2.7/site-packages/gunicorn/arbiter.py", line 345, in halt
    self.stop()
  File "/opt/openlibrary/venv/local/lib/python2.7/site-packages/gunicorn/arbiter.py", line 393, in stop
    time.sleep(0.1)
  File "/opt/openlibrary/venv/local/lib/python2.7/site-packages/gunicorn/arbiter.py", line 245, in handle_chld
    self.reap_workers()
  File "/opt/openlibrary/venv/local/lib/python2.7/site-packages/gunicorn/arbiter.py", line 525, in reap_workers
    raise HaltServer(reason, self.WORKER_BOOT_ERROR)
gunicorn.errors.HaltServer: <HaltServer 'Worker failed to boot.' 3>
```

**Comment by cdrini** [comment]
Fixed by reverting to previous deploy on ol-covers1, and restarting with:

```sh
sudo supervisorctl status
```

and then restart both:

```sh
sudo supervisorctl restart BLAH
```

**Comment by cdrini** [comment]
@cclauss Any thoughts on what might've caused this one?

**Comment by cdrini** [comment]
We updated `six` on `ol-covers1` from 1.11 -> 1.15 (as listed in requirements_common), but the problem persisted.

**Comment by cclauss** [comment]
This exception is raised when pip is <= 1.3.0...
```
python2 -m pip install six==1.3.0
python2 -c "from six.moves.urllib.parse import splitquery, unquote, unquote_plus"
# --> ImportError: No module named urllib.parse
```

**Comment by mekarpeles** [comment]
Many changes in this PR use `requests` which I thought may be a smoking-gun. We attempted to (in virtualenv on ol-covers1) upgrade to the version pinned in requirements.txt and we got a message about `requests` upgrade resulting in `six` upgrades that may be incompatible with `sixpack-client`. So, we upgraded `sixpack-client` and then `requests`, updated our `openlibrary` symlinks and then restarted the service and things worked successfully.

```
sudo -u openlibrary pip install sixpack-client==1.2.0
sudo -u openlibrary pip install requests==2.22.0
```
